api/v1beta1/network_disruption.go

```diff
@@ -458,3 +454,46 @@ func (h NetworkDisruptionHostSpec) Validate() error {
 
 	return nil
 }
+
+func (s NetworkDisruptionServiceSpec) ExtractAffectedPortsInServicePorts(k8sService *v1.Service) ([]v1.ServicePort, []NetworkDisruptionServicePortSpec) {
+	if len(s.Ports) == 0 {
+		return k8sService.Spec.Ports, nil
+	}
+
+	servicePortsDic := map[string]v1.ServicePort{}
+	goodPorts, notFoundPorts := []v1.ServicePort{}, []NetworkDisruptionServicePortSpec{}
+
+	// Convert service ports from found k8s service to a dictionary in order to facilitate the filtering of the ports
+	for _, port := range k8sService.Spec.Ports {
+		servicePortsDic[strconv.Itoa(int(port.Port))] = port
+		if port.Name != "" {
+			servicePortsDic[port.Name] = port
+		}
+	}
+
+	for _, allowedPort := range s.Ports {
+		if allowedPort.Port != 0 {
+			servicePort := servicePortsDic[strconv.Itoa(allowedPort.Port)]
+
+			if servicePort.Port == 0 || (allowedPort.Name != "" && allowedPort.Name != servicePort.Name) {
+				notFoundPorts = append(notFoundPorts, allowedPort)
+
+				continue
+			}
```

Comment:
```suggestion
			servicePort, ok := servicePortsDic[fmt.Sprintf("port-%d", allowedPort.Port)]

			if !ok || (allowedPort.Name != "" && allowedPort.Name != servicePort.Name) {
				notFoundPorts = append(notFoundPorts, allowedPort)

				continue
			}
```

I'm not sure about discarding a port that is found TBH 🤔 (found by port number, but not matching name)

What is your reasoning/feeling that led you to do this?

Reply:
I feel like being strict in the way we find things will eliminate possible user errors, so I'd like the user provided fields to be exact. If the user is misnaming their servicePort I don't want them to be able to use this port in case it's actually the other one they wanted to use. I'm not that set on it though, if you strongly believe it would be best to allow the user to disrupt the provided port even if the provided name does not match then I can change the behavior.

Reply:
With the documentation you added it's pretty clear and this behaviour seems right to me.

Thanks for the explanation, it definitely makes sense for me also now 🤗